src/bun.js/readline.exports.js

```diff
@@ -0,0 +1,3122 @@
+// Attribution: Some parts of of this module are derived from code originating from the Node.js
+// readline module which is licensed under an MIT license:
+//
+// Copyright Node.js contributors. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+// ----------------------------------------------------------------------------
+// Section: Imports
+// ----------------------------------------------------------------------------
+var { Array, RegExp, String, Bun } = import.meta.primordials;
+var EventEmitter = import.meta.require("node:events");
+var { clearTimeout, setTimeout } = import.meta.require("timers");
+var { StringDecoder } = import.meta.require("string_decoder");
+
+var { inspect } = Bun;
+var debug = process.env.DEBUG ? console.log : () => {};
```

Comment:
We _probably_ should switch to a more specific DEBUG flag since this is also used by the `debug` npm package

Reply:
Yeah I was thinking about that too. Should there be a bun wide debug or a debug flag for difference modules? `BUN_DEBUG`?

Reply:
I renamed it to `BUN_JS_DEBUG` for now. Let me know if you think that is too vague or too specific